component/src/main/java/io/siddhi/extension/io/gcs/sink/GCSSink.java

```diff
@@ -0,0 +1,334 @@
+                        optional = true,
+                        defaultValue = "EMPTY_STRING",
+                        description = "Access Control List for the bucket level ACL defined as a key value pair list" +
+                                " defined as \"'<key>:<value>','<key>:<value>'\""
+                ),
+                @Parameter(
+                        name = GCSConstants.OBJECT_NAME,
+                        type = DataType.STRING,
+                        dynamic = true,
+                        description = "Full name of the object given to the object including the path"
+                ),
+                @Parameter(
+                        name = GCSConstants.FLUSH_SIZE,
+                        type = DataType.INT,
+                        optional = true,
+                        defaultValue = "1",
+                        description = "Number of events that the sink will wait before making a file commit"
+                ),
+                @Parameter(
+                        name = GCSConstants.ROTATE_INTERVAL_MS,
+                        type = DataType.INT,
+                        optional = true,
+                        defaultValue = "-1",
+                        description = "Maximum Time Span an Object should be kept open to add events before writing a" +
+                                " file commit in milliseconds, This option cannot be used with when flush.size is" +
+                                " enabled"
+                ),
+                @Parameter(
+                        name = GCSConstants.ENCLOSING_ELEMENT,
+                        type = DataType.STRING,
+                        optional = true,
+                        defaultValue = GCSConstants.DEFAULT_ENCLOSING_ELEMENT,
+                        description = "Enclosing element to contain the events in case an xml mapper is used"
+                ),
+                @Parameter(
+                        name = GCSConstants.TEXT_DELIMITER,
+                        type = DataType.STRING,
+                        optional = true,
+                        defaultValue = GCSConstants.DEFAULT_TEXT_DELIMITER,
+                        description = "Delimiter to be used as event separator when text/binary mapper is used"
+                ),
+                @Parameter(
+                        name = GCSConstants.FLUSH_TIMEOUT,
+                        type = DataType.INT,
+                        optional = true,
+                        defaultValue = GCSConstants.DEFAULT_FLUSH_TIMEOUT,
+                        description = "Timeout that the sink should wait before making a file commit when Flush " +
+                                "Size option is enabled"
+                )
+        },
+        examples = {
+                @Example(
+                        syntax = "@sink(type='google-cloud-storage'," +
+                                "credential.provider.file.path='<Auth.file.path>'," +
```

Comment:
Shall we use a common name like in https://github.com/siddhi-io/siddhi-io-googlepubsub/blob/master/component/src/main/java/io/siddhi/extension/io/googlepubsub/util/GooglePubSubConstants.java#L29